pkg/cloudprovider/cloudprovider.go

```diff
@@ -337,3 +338,22 @@ func ParseResourceID(resourceID string) (azure.Resource, error) {
 
 	return result, nil
 }
+
+// recordError records the error in appropriate metric
+func recordError(reporter *metrics.Reporter, operation string) {
```

Comment:
There is a weird behavior with this metric. On the one hand, when looking on `aadpodidentity_nodemanagedidentity_operations_duration_seconds_bucket` metric, we can see a lot of requests (with operation_type=`/metadata/identity/oauth2/token`) with status_code= 403. On the other hand, we have no data at all from `aadpodidentity_cloud_provider_operations_errors_count` metric. Would appreciate your help,

Reply:
I understand that part of this 403 is wired noise, but I think it is very important to be able to get "quality" 403, without a noise - should allow to monitor and rise alert.

Reply:
@DanielShor , @elutsky - 
As per my understanding following are the scenarios where NMI will return 403.

1. Failed to acquire a token using the MSI
2. No matched identity in CREATED state was found
3. Request blocked by AAD Pod Identity NMI

"No data at all from aadpodidentity_cloud_provider_operations_errors_count metric"
Only (1) from above 'Failed to acquire token via MSI', falls in the bucket of cloud_provider_operations and is not being tracked.
I will take up task of adding MSI token operations to cloud_provider_operations metric.

Thank you for finding this gap and bringing to our notice.
@kkmsft / @aramase - Any other thoughts on this ?

Reply:
@DanielShor , @elutsky , @kkmsft / @aramase - Added MSI token operations to cloud_provider_operations metric